### ISSUE TITLE
Discourse Integration app don’t redirect me back to Nextcloud and don’t work

### ISSUE THREAD

**Issue by LosHunterros** [nextcloud/integration_discourse#61]
Nextcloud version (eg, 20.0.5) : 29.0.0
Operating system and version (eg, Ubuntu 20.04) : Ubuntu 20.04.6 LTS
Apache or nginx version (eg, Apache 2.4.25) : Apache/2.4.41
PHP version (eg, 7.4) : 8.2

I am trying to connect Nextcloud with Discourse by Discourse Integration app. This is my configuration in Discourse:
![image](https://github.com/nextcloud/integration_discourse/assets/123071857/4970c89b-374c-4533-9145-25091bc38e49)

And in Nextcloud:
![image](https://github.com/nextcloud/integration_discourse/assets/123071857/45a49d45-fac2-49da-8dd6-f6fd515d8a61)

When i click on ‘Connect to Discourse’ i am redirected to Discourse:
![image](https://github.com/nextcloud/integration_discourse/assets/123071857/cabd575c-4e66-4d6c-a062-4b20d7a92c07)

But if i click ‘Authorize’ i get an error, i am not redirected back to Nextcloud and i don’t have any changes in Nextcloud
![image](https://github.com/nextcloud/integration_discourse/assets/123071857/948298ae-a0f6-4605-a599-17748ed18629)
![image](https://github.com/nextcloud/integration_discourse/assets/123071857/e22cf87f-64e3-445f-85e9-e4365c1c3bbd)

Error:
Failed to launch 'web+nextclouddiscourse://auth-redirect?payload=h0zAd7muVMb9BOgGy7h9yAHjlfo%2F4H3uW%2BerBcTqjtuI0cAE2i8%2F758rjIJJ%0AZICHZUdkEd2yokRhJZBdY1XuHMdwhfH%2Bixsq%2FJjwp9BI2ptxAY6ZqffmMShm%0APY8dcr2GFCnrJ9Cvbeu1EqbyQ0M0jarTBkphJ4nOeoTkuKNbQJhDkbmDwhup%0A%2Benz%2Bkyn43ddT4GulLXBtl0ZhzbfR0kkkwbnGJZoGWsLBXmWr%2FyltCve4dEI%0AlQZKWexlnasa%2B0hx6gq7C9mx9fT5viVLBBG2RlGGTACI7zppxXAbnmKTebw%2B%0AoRVYJXIAFnboHsr0WbsI65ixJLmj0KccvCBaBssAsA%3D%3D%0A
’ because the scheme does not have a registered handler.

What am i doing wrong?

P.S.
If that’s important i have Discourse SSO app installed and working properly.

**Comment by keithkhl** [comment]
I have exactly the same problem, just with different language.
I've also played around with api key, but nothing worked so far.

**Comment by andrey18106** [comment]
@LosHunterros Hello, have you configured the Application (should be prompted like described on settings page) in your browser to handle such links to redirect back to correct instance URL? Have you tried another browser?

**Comment by LosHunterros** [comment]
You're right. Currently i am able to do this only in Firefox. Other browsers don't show any prompt so i don't know how to accept this - i don't know - custom protocol? But Firefox handle everything fine.

**Comment by andrey18106** [comment]
> You're right. Currently i am able to do this only in Firefox. Other browsers don't show any prompt so i don't know how to accept this - i don't know - custom protocol? But Firefox handle everything fine.

There is `registerProtocolHandler` used to do so, and it is not supported only in Safari, additionally it works only in secure context (https). https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler#browser_compatibility

If you are using chromium based browser, could you please double check if enabling `chrome://settings/handlers` -> `Allow sites to ask to become default handlers for protocols` resolves the problem and if there are already registered entries? 

Anyway there will be a button to trigger protocol registration manually in the next update.

**Comment by LosHunterros** [comment]
It looks like this:

https://github.com/user-attachments/assets/81c4c03b-21a3-4572-b6a6-0743ee2fc889
